app/src/main/java/com/asfoundation/wallet/ui/Erc681ReceiverPresenter.kt

```diff
@@ -0,0 +1,74 @@
+package com.asfoundation.wallet.ui
+
+import android.os.Bundle
+import com.appcoins.wallet.bdsbilling.WalletService
+import com.asfoundation.wallet.entity.TransactionBuilder
+import com.asfoundation.wallet.service.WalletGetterStatus
+import com.asfoundation.wallet.ui.iab.InAppPurchaseInteractor
+import com.asfoundation.wallet.util.TransferParser
+import io.reactivex.Observable
+import io.reactivex.Scheduler
+import io.reactivex.disposables.Disposable
+
+internal class Erc681ReceiverPresenter(private val view: Erc681ReceiverView,
+                                       private val transferParser: TransferParser,
+                                       private val inAppPurchaseInteractor: InAppPurchaseInteractor,
+                                       private val walletService: WalletService,
+                                       private val data: String,
+                                       private val viewScheduler: Scheduler) {
+  private var disposable: Disposable? = null
+  fun present(savedInstanceState: Bundle?) {
+    if (savedInstanceState == null) {
+      disposable =
+          handleWalletCreationIfNeeded().takeUntil {
+            it != WalletGetterStatus.CREATING.toString()
+          }
```

Comment:
for readability i think we should change to   
handleWalletCreationIfNeeded()
              .takeUntil { it != WalletGetterStatus.CREATING.toString() }